cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedWorkflowProgramRunner.java

```diff
@@ -66,14 +81,23 @@ protected ProgramController launch(Program program, ProgramOptions options,
     WorkflowSpecification workflowSpec = appSpec.getWorkflows().get(program.getName());
     Preconditions.checkNotNull(workflowSpec, "Missing WorkflowSpecification for %s", program.getName());
 
-    // Localize the spark-assembly jar, since workflow can execute spark.
-    File sparkAssemblyJar = SparkUtils.locateSparkAssemblyJar();
-    localizeFiles.put(sparkAssemblyJar.getName(), sparkAssemblyJar);
+    // It the workflow has Spark, localize the spark-assembly jar
+    List<String> extraClassPaths = new ArrayList<>();
```

Comment:
We are switching to Java7 syntax, which is more compact and better be using language feature.